File: app/src/main/java/com/luseen/ribble/presentation/HomeActivity.java
package com.luseen.ribble.presentation;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.luseen.logger.Logger;
import com.luseen.ribble.App;
import com.luseen.ribble.R;
import com.luseen.ribble.data.entity.Shot;
import com.luseen.ribble.data.network.Api;
import com.luseen.ribble.data.network.ApiService;
import com.luseen.ribble.data.pref.Preferences;
import com.luseen.ribble.presentation.base.BaseActivity;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

public class HomeActivity extends BaseActivity {

    @Inject
    protected ApiService apiService;

    @Inject
    protected Preferences preferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        getAppComponent().inject(this);

        Logger.log(apiService);
        Logger.log(preferences);
        apiService.getShots(20,"c14f402db06bc15be330b7a4fd049d1ca88fe5fa23892c073ca97dc422cfe9ee")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<List<Shot>>() {
                    @Override
                    public void accept(List<Shot> o) throws Exception {
                        Logger.log(o.size());
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        Logger.log(throwable);
                    }
                });
    }
}

File: app/src/main/java/com/luseen/ribble/presentation/screen/home/HomeActivity.kt
package com.luseen.ribble.presentation.screen.home

import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.v4.app.FragmentManager
import android.support.v4.view.GravityCompat
import android.view.MenuItem
import com.luseen.ribble.R
import com.luseen.ribble.presentation.base_mvp.base.BaseActivity
import com.luseen.ribble.presentation.screen.shot_root.ShotRootFragment
import kotlinx.android.synthetic.main.activity_home.*
import kotlinx.android.synthetic.main.app_bar_main.*
import javax.inject.Inject

class HomeActivity : BaseActivity<HomeContract.View, HomeContract.Presenter>(), HomeContract.View,
        NavigationView.OnNavigationItemSelectedListener {

    @Inject
    protected lateinit var homePresenter: HomePresenter

    @Inject
    protected lateinit var fragmentManager: FragmentManager

    override fun initPresenter(): HomeContract.Presenter = homePresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        initViews()
    }

    private fun initViews() {
        setSupportActionBar(toolbar)
//        val toggle = ActionBarDrawerToggle(
//                this, drawerLayout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        //drawerLayout.addDrawerListener(toggle)
        //toggle.syncState()
        supportActionBar?.setDisplayShowTitleEnabled(false)
        navView.setNavigationItemSelectedListener(this)
    }

    override fun injectDependencies() {
        activityComponent.inject(this)
    }

    override fun openShotFragment() {
        fragmentManager
                .beginTransaction()
                .add(R.id.container, ShotRootFragment.newInstance())
                .commit()
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_camera -> {

            }
            R.id.nav_gallery -> {

            }
            R.id.nav_slideshow -> {

            }
            R.id.nav_manage -> {

            }
            R.id.nav_share -> {

            }
            R.id.nav_send -> {

            }
        }

        drawerLayout.closeDrawer(GravityCompat.START)
        return true
    }
}
